### PR TITLE
shared-mime-info: add patch for Objective C++

### PR DESCRIPTION
**Pull request by mkae**
- remove Id keyword
- take maintainership
- set env var XDG_DATA_DIRS when running update-mime-database for later

### CODE REVIEW

File: devel/shared-mime-info/Portfile [hunk 1]
```diff
@@ -40,12 +44,15 @@ configure.cmd       ./autogen.sh
 
 use_parallel_build  no
 
-configure.args      --disable-silent-rules \
-                    --disable-update-mimedb
+configure.args      --disable-update-mimedb
```
Comment (larryv) [inline devel/shared-mime-info/Portfile]: Why did you remove `--disable-silent-rules`?

**Comment by RJVB** [comment]
This is one of the ports I've been running for quite a while now with these modifications and that Marko and I are introducing in the context of the qt5-kde and afterwards the KF5 introduction.

I think the accurate question is why David added the option a month or so ago. I either missed it when I synced back then, or else I upgraded my own version independently. It's not a big deal to add the option, but does it really teach us anything of interest to have (even) more verbose build output?

```
> configure --help
<snip>
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
```

(FWIW, this is something where I'm inclined to think that the maintainer would add the flag or else add V=1 via one of the build.*args options temporarily, in case of some kind of build issue.)

**Comment by mkae** [comment]
So, I'll reinsert it. OK.

**Comment by larryv** [comment]
Logs are cheap. We prefer that they contain as much information as possible so that we don't have to tell users to go back and reproduce the problem.

**Comment by mkae** [comment]
@larryv, I agree!

**Comment by larryv** [comment]
Has this been submitted upstream?

**Comment by RJVB** [comment]
On Wednesday November 16 2016 16:25:08 Lawrence Vel�zquez wrote:
>Has this been submitted upstream?

No, I've been putting that off but it woud certainly be an option once it's seen more testing in the wild.

File: devel/shared-mime-info/files/ObjCpp.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+It comes with ABSOLUTELY NO WARRANTY, to the extent permitted by law. You may
+redistribute copies of update-mime-database under the terms of the GNU General
+Public License. For more information about these matters, see the file named
+COPYING.
+-->
+<!--
+Notes:
+- the mime types in this file are valid with the version 0.30 of the
```
Comment (mkae) [inline devel/shared-mime-info/files/ObjCpp.xml]: Version 0.3?

**Comment by RJVB** [comment]
On Thursday November 17 2016 00:56:05 Marko K�ning wrote:

>+- the mime types in this file are valid with the version 0.30 of the
>Version 0.3?

Good question; see below.

Re: my earlier answer of submitting this upstream: I'm not actually sure where I got this file from, if I found it online or created it myself from a preexisting ObjC entry in a preexisting other XML file. Judging from that version statement above this would have to be kde.xml which is provided by kdelibs.
Makes it a bit delicate to submitting a patch upstream... and where.

In fact, shared-mime-info intends to be extendable via the mime/packages directory, so it probably isn't necessary to submit this upstream. 

I see I do have to check if there is an equivalent to kde.xml shipping with KF5. If not we will have to consider including an ObjC entry with shared-mime-info.

**Comment by mkae** [comment]
If it originates from KDE shouldn't that be said so in the license header?

**Comment by RJVB** [comment]
On Thursday November 17 2016 07:32:05 Marko K�ning wrote:
> If it originates from KDE shouldn't that be said so in the license header?

I've maintained the copyright header, and in any case kde.xml does *not* have an ObjC++ entry so KDE cannot have a copyright on it ;)

R>

**Comment by mkae** [comment]
Ah, okay, I mistakenly thought you copied it from them. Sorry for the noise. :-)

**Comment by larryv** [comment]
MacPorts is not your patch-testing laboratory. We don't want to keep your feature patches around forever. Please submit this upstream.

**Comment by mkae** [comment]
You have a point, @larryv!

**Comment by RJVB** [comment]
You know what Marko, I don't need "their" permission to run my own patches nor instructions what and when I submit upstream so let's just drop the thing and be done with it.

**Comment by mkae** [comment]
Cool it, guys. We'll get this done together! :-)

Will sign these changes off and commit it soonish.

Thanks for all the patience here!

**Comment by mkae** [comment]
@larryv, are you pushing your changes to this PR, or did I misunderstand my PM?

**Comment by mkae** [comment]
@larryv, BTW, did you split up this submission manually or did you use some tool to end up with 4 logically clean commits instead of a single squashed one?

**Comment by larryv** [comment]
@mkae I did it manually with `git reset` and a `git add`/`git commit` cycle.

I haven't pushed my revision yet; I'll do that within the next 6 hours.

**Comment by mkae** [comment]
@larryv, thanks big time! 👍

**Comment by larryv** [comment]
The tweaks to the portfile are not really relevant here, so I broke those off and pushed them to master as separate commits (101b66335ed0121b0ef49f6a50db2b59d85cbc3e, 8fd4fedf18db966aedb1450f62b3092aa571a5c2, d0a125505343b650472fdf74ca7cab18b7ce94eb). Only the new XML file remains.

I reiterate that, under normal circumstances, we should not be maintaining new features above and beyond what upstream provides. I would still like to see some effort to incorporate this upstream, like a bug report or mailing list post. If they reject it for some reason, then we can certainly consider maintaining it ourselves.

**Comment by mkae** [comment]
Yes, @larryv, let's try to go this route and contact upstream.

**Comment by RJVB** [comment]
https://bugs.freedesktop.org/show_bug.cgi?id=98823

**Comment by larryv** [comment]
Thanks for the upstream report; I appreciate it.
